File: job-agent-os/frontend/src/types/approval.ts
/** Approval types (matches backend schemas/approval.py) */

export type ApprovalAction = "approve" | "reject" | "modify" | "skip";

export interface ApprovalResponse {
  id: string;
  approval_type: string;
  title: string;
  description: string | null;
  payload: Record<string, unknown>;
  options: string[];
  status: string;
  timeout_seconds: number | null;
  requested_at: string;
}

export interface ApprovalRespondRequest {
  action: ApprovalAction;
  feedback?: string | null;
  modified_payload?: Record<string, unknown> | null;
}

File: job-agent-os/src/job_agent_os/schemas/approval.py
"""Approval schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ApprovalResponse(BaseModel):
    """Approval response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    approval_type: str
    title: str
    description: str | None = None
    payload: dict
    options: list[str] = Field(default_factory=list)
    status: str
    action: str | None = None
    feedback: str | None = None
    timeout_seconds: int | None = None
    requested_at: datetime
    responded_at: datetime | None = None


class ApprovalRespondRequest(BaseModel):
    """Approval respond request."""

    action: str = Field(pattern="^(approve|reject|modify|skip)$")
    feedback: str | None = None
    modified_payload: dict | None = None


class BatchApprovalItem(BaseModel):
    """Batch approval item."""

    approval_id: UUID
    action: str
    feedback: str | None = None


class BatchApprovalRequest(BaseModel):
    """Batch approval request."""

    approvals: list[BatchApprovalItem]
